lib/Doctrine/ORM/Query/Parameter.php

```diff
@@ -67,6 +69,10 @@ class Parameter
      */
     public function __construct($name, $value, $type = null)
     {
+        if (preg_match('/^:|:$/', $name)) {
+            @trigger_error('Starting or ending a parameter name with ":" is deprecated since 2.7 and will cause an error in 3.0', E_USER_DEPRECATED);
```

Comment:
Let's add the package name and use some linebreaks:
```suggestion
            @trigger_error(
                'Starting or ending a parameter name with ":" is deprecated since Doctrine ORM 2.7 and will cause an error in 3.0',
                E_USER_DEPRECATED
            );
```